libyul/optimiser/SyntacticalEquality.cpp

```diff
@@ -181,6 +181,11 @@ bool SyntacticallyEqual::statementEqual(Block const& _lhs, Block const& _rhs)
 	});
 }
 
+bool SyntacticallyEqual::statementEqual(Comment const& _lhs, Comment const& _rhs)
+{
+	return _lhs.text == _rhs.text;
```

Comment:
I think we should just ignore comments here - otherwise we cannot unify
```
function f() -> x {
   // foo
  x := 2
}
function g() -> x {
  // bar
  x := 2
}
```

Reply:
You are right, Comments are unrelated to syntax. Good point (we could make it a boolean defaulting to false, to indicate whether comments should be taken into account, too.)